packages/cli/src/utils/testRunner.ts

```
@@ -232,6 +266,92 @@
+            let baseName = options.filter;
+            if (baseName.endsWith('.test.ts') || baseName.endsWith('.test.js')) {
+              baseName = baseName.slice(0, -8); // Remove '.test.ts' or '.test.js'
+            } else if (baseName.endsWith('.test')) {
+              baseName = baseName.slice(0, -5); // Remove '.test'
+            }
+
+            // Now match against filename or suite name with the processed filter
+            const matchesFileName = fileNameWithoutExt.includes(baseName);
+            const matchesSuiteName = testSuite.name && testSuite.name.includes(baseName);
+
+            if (!matchesFileName && !matchesSuiteName) {
```

Biome (1.9.4)

[error] Change to an optional chain.

Unsafe fix: Change to an optional chain.


(lint/complexity/useOptionalChain)